vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -422,9 +423,10 @@ def _collect_mors_and_attributes(self, server_instance):
                 for prop in obj.missingSet:
                     error_counter += 1
                     self.log.error(
-                        "Unable to retrieve property {} for object {}: {}".format(
-                            ensure_unicode(prop.path), ensure_unicode(obj.obj), ensure_unicode(prop.fault)
-                        )
+                        "Unable to retrieve property %s for object %s: %s",
```

Comment:
Are all of these really fixing encoding issues, or just kind of hiding them because we don't format the string if we're not logging ?

Reply:
No there is a behavior difference:
```
>>> "foo %s" % u"é"
u'foo \xe9'
>>> "foo {}".format(u"é")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 0: ordinal not in range(128)
```

Reply:
This example is not valid for the check, because we use `from __future__ import unicode_literals`
```>>> from __future__ import unicode_literals
>>> "foo %s" % u"é"
u'foo \xe9'
>>> "foo {}".format(u"é")
u'foo \xe9'
>>>```

Reply:
In the check we should then be manipulating only unicode objects since we're using `ensure_unicode` on every variable when formatting strings. So I'm not sure how this solves anything

Reply:
OK. Maybe it's not the root of the issue then. I'm sure it's solves something because master doesn't work in the environment, and this does.

Reply:
OK.
I'd have liked to know for a fact what was failing, but the changes are fine anyway so i'm approving

Reply:
I tested again, and you're right, the issue was not logging, but get_agg_key.